dlt/destinations/weaviate/weaviate_client.py

```diff
@@ -0,0 +1,540 @@
+            additional_headers=config.credentials.additional_headers,
+        )
+
+    def make_full_name(self, table_name: str) -> str:
+        """Make a full Weaviate class name from a table name by prepending
+        the dataset name if it exists.
+        """
+        return (
+            f"{self.config.dataset_name}_{table_name}"
+            if self.config.dataset_name
+            else table_name
+        )
+
+    def get_class_schema(self, table_name: str) -> Dict[str, Any]:
+        """Get the Weaviate class schema for a table."""
+        return self.db_client.schema.get(self.make_full_name(table_name))
+
+    def create_class(self, class_schema: Dict[str, Any]) -> None:
+        """Create a Weaviate class."""
+
+        updated_schema = class_schema.copy()
+        updated_schema["class"] = self.make_full_name(updated_schema["class"])
+
+        self.db_client.schema.create_class(updated_schema)
+
+    def create_class_property(
+        self, table_name: str, prop_schema: Dict[str, Any]
+    ) -> None:
+        """Create a Weaviate class property."""
+        self.db_client.schema.property.create(
+            self.make_full_name(table_name), prop_schema
+        )
+
+    def delete_class(self, table_name: str) -> None:
+        """Delete a Weaviate class."""
+        self.db_client.schema.delete_class(self.make_full_name(table_name))
+
+    def query_class(self, table_name: str, properties: List[str]) -> GetBuilder:
+        """Query a Weaviate class."""
+        return self.db_client.query.get(self.make_full_name(table_name), properties)
+
+    def create_object(self, obj: Dict[str, Any], table_name: str) -> None:
+        """Create a Weaviate object."""
+        self.db_client.data_object.create(obj, self.make_full_name(table_name))
+
+    @wrap_weaviate_error
+    def initialize_storage(self, truncate_tables: Iterable[str] = None) -> None:
+        if not self.is_storage_initialized():
+            self.create_sentinel_class()
+        elif truncate_tables:
+            for table_name in truncate_tables:
+                try:
+                    class_schema = self.get_class_schema(table_name)
+                except weaviate.exceptions.UnexpectedStatusCodeException as e:
+                    if e.status_code == 404:
+                        continue
+                    raise
+
+                self.delete_class(table_name)
+                self.create_class(class_schema)
+
+    @wrap_weaviate_error
+    def is_storage_initialized(self) -> bool:
+        try:
+            self.db_client.schema.get(self.config.dataset_name)
+        except weaviate.exceptions.UnexpectedStatusCodeException as e:
+            if e.status_code == 404:
+                return False
+            raise
+        return True
+
+    def create_sentinel_class(self) -> None:
+        self.db_client.schema.create_class(
+            {
+                "class": self.config.dataset_name,
```

Comment:
here's the same situation. my take: just like in the sql_client, add a method to get `dataset_name` that will normalize it etc.